File: Python/Math/gcd_&_lcm/Replace Non-Coprime Numbers in Array.py
class Solution(object):
    def gcd(self,a,b):
        if b==0:
            return a
        return self.gcd(b,a%b)
    def replaceNonCoprimes(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        st=[]
        for num in nums:
            st.append(num)
            while len(st)>1:
                a,b=st[-2],st[-1]
                g = self.gcd(a, b)
                if g>1:
                    st.pop()
                    st.pop()
                    lcm=a*b//g
                    st.append(lcm)
                else:
                    break
        return st

File: CPP/Math/gcd_&_lcm/Replace Non-Coprime Numbers in Array.cpp
class Solution {
public:
    vector<int> replaceNonCoprimes(vector<int>& nums) {
        vector<int> st;
        for (auto num : nums) {
            st.push_back(num);
            while (st.size() > 1) {
                int a = st[st.size() - 1];
                int b = st[st.size() - 2];
                int g = gcd(a, b);
                if (g > 1) {
                    st.pop_back();
                    st.pop_back();
                    long long lcm = (1LL * a * b) / g;
                    st.push_back((int)lcm);
                } else {
                    break;
                }
            }
        }
        return st;
    }
};
